src/pip/_internal/commands/install.py

```diff
@@ -528,14 +523,24 @@ def _warn_about_conflicts(self, to_install):
                 )
 
 
-def get_lib_location_guesses(*args, **kwargs):
-    scheme = distutils_scheme('', *args, **kwargs)
+def get_lib_location_guesses(
+        user=False,  # type: bool
+        home=None,  # type: Optional[str]
+        root=None,  # type: Optional[str]
+        isolated=False,  # type: bool
+        prefix=None  # type: Optional[str]
+):
+    # type:(...) -> List[str]
+    scheme = distutils_scheme('', user=user, home=home, root=root,
+                              isolated=isolated, prefix=prefix)
     return [scheme['purelib'], scheme['platlib']]
 
 
-def site_packages_writable(**kwargs):
+def site_packages_writable(root, isolated):
+    # type: (Optional[str], bool) -> bool
```

Comment:
Shouldn't it be like `site_packages_writable(isolated, root=None)`? As `root` can be optional, but since it'd be provided anyway, So this should be okay too. :D

Reply:
I actually followed the pattern of arguments in [pip._internal.locations::distutils_scheme](https://github.com/pypa/pip/blob/master/src/pip/_internal/locations.py#L95)

Also the function [right above it](https://github.com/pypa/pip/pull/8365/files#diff-680934feb85d0d4680a574cfd5d1da14L531) has the default value of `root=None`, so IMO we can avoid instantiating `root=None` here.

But I don't have a strong opinion for it, but I think if we want to change this, 
`site_packages_writable(root=None, isolated=False)` is better